### PR TITLE
Homework1

### PR DESCRIPTION
**Pull request by SargisV**
Haven't finished working with the 1st task's second solution

### CODE REVIEW

**Comment by SargisV** [comment]
sorry for some unnecessary staff that pull request contains

File: homeworks/sargis_vardanyan/homework1/bitwiseop/BitwiseOperators.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package bitwiseop;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+public class BitwiseOperators {
+    //Task https://www.hackerrank.com/challenges/bitwise-operators-in-c/problem?h_r=internal-search&isFullScreen=true
```
Comment (sirak92) [inline homeworks/sargis_vardanyan/homework1/bitwiseop/BitwiseOperators.java:8]: it's great that you have link here, but you can already get yourself familiar with javadoc comments and you javadoc format to add links

File: homeworks/sargis_vardanyan/homework1/bitwiseop/BitwiseOperators.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    //Simple solution using for. Time complexity O(n^2)
+    public static List<Integer> calcTheMax(int n, int k) {
+        int maxAnd = 1 & n;
+        int maxOr = 1 | n;
+        int maxXor = 1 ^ n;
+
+        for (int i = 0; i <= n; i++) {
+            for (int j = i + 1; j <= n; j++) {
```
Comment (sirak92) [inline homeworks/sargis_vardanyan/homework1/bitwiseop/BitwiseOperators.java:25]: small optimization, i < n

File: homeworks/sargis_vardanyan/homework1/bitwiseop/BitwiseOperators.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        scanner.close();
+
+        System.out.println(calcTheMax(n,k));
+    }
+
+    //Simple solution using for. Time complexity O(n^2)
+    public static List<Integer> calcTheMax(int n, int k) {
+        int maxAnd = 1 & n;
+        int maxOr = 1 | n;
+        int maxXor = 1 ^ n;
```
Comment (sirak92) [inline homeworks/sargis_vardanyan/homework1/bitwiseop/BitwiseOperators.java:22]: you can initialize them with 0 values, no need to do extra calculations

File: homeworks/sargis_vardanyan/homework1/bitwiseop/BitwiseOperators.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        // finding max value of the logical XOR within the range of [1, n]
+        // same way as logical OR
+        int xored = 1 ^ n;
+        int significantBitForXOR = 32 - Integer.numberOfLeadingZeros(xored);
+        int maxXOR = (1 << significantBitForXOR) - 1;
+        
+        // trying to find some connection with these max values(maxAND, maxOR, maxXOR)
+        // and k
+        return null;
+    }
```
Comment (sirak92) [inline homeworks/sargis_vardanyan/homework1/bitwiseop/BitwiseOperators.java:60]: didn't go deep into the algorithm, please remind me to share this solution in the class room

**Review by sirak92 — COMMENTED** [review]
Excellent work Sargis, keep it up!

File: homeworks/sargis_vardanyan/homework1/bitwiseop/BitwiseOperators.java (hunk 4 above)
Comment (SargisV) [reply homeworks/sargis_vardanyan/homework1/bitwiseop/BitwiseOperators.java:62]: this is a bit complicated I guess

**Pull request merged**